2.3 release announcement (#239)
mixxxdj/website: content/news/XXXX-XX-XX-mixxx-2-3-0-released.md

title: Mixxx 2.3.0 released
authors: Jan Holthuis
tags: 2.3, 2.3.0, release announcement
status: draft
comments: yes

> After a long waiting time, we are proud to announce the release of Mixxx 2.3.0, [download it!]({filename}/pages/download.md#stable)

Reviewer: "after a long waiting time"
this is grammatically a little odd in English.  Better: "At last", "After a long time", "After waiting a long time"

For the full list of changes, have a look at the [changelog](https://manual.mixxx.org/2.3/en/chapters/appendix/version_history.html) and the [2.3.0 milestone on Launchpad](https://launchpad.net/mixxx/+milestone/2.3.0).

This has been a rather big release, with lots of useful changes, bug fixes and improvements.
All in all, a total of 7183 changes changed over 1 million lines of code were changed since the [2.2.4 release]({filename}/news/2020-06-11-Mixxx-2-2-4-released.md)!

> **TODO:** Update the numbers above using `git shortlog -sn release-2.2.4..upstream/2.3 | wc -l`, `git log --oneline --no-merges release-2.2.4..upstream/2.3 -- . ':!lib' | wc -l` and `git diff --shortstat release-2.2.4..upstream/2.3 -- . ':!lib`.

Reviewer: does markdown support comments, so that this will be guaranteed not to appear in the final version?

Author: This is not a comment, it's just a reminder to do this after tagging and before merge, and then remove that comment.

Reviewer: yeah I know, I have just seen enough blog posts go live with "TO DO" "TKTKTKTKTK" "USE REAL DATA" that I prefer never to have it in there at all. don't worry about it. (https://stackoverflow.com/questions/4823468/comments-in-markdown)

> Many people spent their free time working on Mixxx and reported bugs, translated Mixxx into other languages, improved the manual, contributed controller mappings, improved skins and hacked on the core code. Thanks a lot to all of them, we really appreciate it!

Reviewer: can we also mention that the manual was updated by our Outreachy Intern?

Author: Sure, do you have a specific proposal?

Reviewer: Many people spent their free time working on Mixxx and reported bugs, translated Mixxx into other languages, contributed controller mappings, improved skins and hacked on the core code. This release also features improvements to our manual made by our first intern from the Outreachy program. Thanks a lot to all our contributors, we really appreciate it!

The Mixxx code (including skins and controller mappings) has received contributions by almost 100 people:

    $ git shortlog -sn release-2.2.4..upstream/2.3
    2305  Jan Holthuis
    1838  Uwe Klotz
    1393  Daniel Schürmann
    1100  ronso0
     899  Be
     264  Owen Williams
     144  Nino Miškić-Pletenac
     131  Ferran Pujol Camins
     118  RJ Skerry-Ryan
      81  Evan Dekker
      79  Harshit Maurya
      71  Swiftb0y
      65  JoergAtGithub
      63  S.Brandt
      57  Kerrick Staley
      52  Daniel Poelzleithner
      52  Matthew Nicholson
      50  Justin Kourie
      49  Christian

>       36  JosepMaJAZ

Reviewer: josep is here under multiple names.  We need to dedupe

Author: I don't know. I thought by just adding the shortlog, we'd have a reproducible, simple way to show the list of contributors. I think it's okay like this. The alternative would be to take the contributor list from the about dialog.

Reviewer: The shortlog is fine I guess, but maybe remove the commit counts? up to you.

Author: Done.

title: Mixxx 2.3.0 released
authors: Jan Holthuis
tags: 2.3, 2.3.0, release announcement
status: draft
comments: yes

> At last, we are proud to announce the release of Mixxx 2.3.0, [download it!]({filename}/pages/download.md#stable)

Reviewer 2: "At last"  feels a bit misplaced as a first sentences. Maybe we can come up with a more catchy phrase and move  this to a the second sentences.

> At last, we are proud to announce the release of Mixxx 2.3.0, [download it!]({filename}/pages/download.md#stable)

Reviewer 2: "download it!" sound not that polite. Can we move it to the end and replace it with "Check it out" or "Give it a spin" or such. Here it might be sufficient to make the version number a download link. Alternative we may add a full sentences something like "You can download it for W/L/M ..."

> This release brings [hotcue colors & labels]({filename}/news/2020-08-25-new-in-2-3-hotcue-colors.md), import of [Serato](2021-02-08-new-in-2-3-serato-support.md) and [Rekordbox]({filename}/news/2020-07-20-new-in-2-3-rekordbox-support.md) libraries, [more accurate key detection]({filename}/news/2021-04-08-new-in-2-3-keyfinder.md), [intro/outro sections]({filename}/news/2020-07-09-intro-outro-sections.md), a revamped LateNight skin, multithreaded analysis, [deck cloning]({filename}/news/2020-11-15-new-in-2-3-deck-clone.md), [track color support]({filename}/news/2020-10-24-new-in-2-3-track-colors.md).

Reviewer 2: This sentences is hard to read. Did you consider to form a bullet list? 

If we can effort a longer post, we may add paragraphs for related features with a sentence that gives some context. 
I can propose some if this is desired.

Author: Rewrote it as text.

> We also added support for recording/streaming in the [Opus codec](https://opus-codec.org/), polished the preferences (including the controller workflow), added some library improvements and [switched the CMake build system generator]({filename}/news/2020-12-14-scons-cmake-migration.md).

Reviewer 2: ```suggestion
We also added support for recording/streaming in the [Opus codec](https://opus-codec.org/), polished the preferences (including the controller workflow), added some library improvements and [switched to the CMake build system generator]({filename}/news/2020-12-14-scons-cmake-migration.md).
```

title: Mixxx 2.3.0 released
authors: Jan Holthuis
tags: 2.3, 2.3.0, release announcement
status: draft
comments: yes

We are proud to announce the release of Mixxx 2.3.0!

> If you want to give it a spin, you can now [download it]({filename}/pages/download.md#stable) on Windows, macOS and Linux.

Reviewer 2: Can we remove the it from the link, that looks too much like a order to me. 
```suggestion
If you want to give it a spin, you can now [download]({filename}/pages/download.md#stable) it on Windows, macOS and Linux.
```

Author: Then the download is a bit unobvious. How about:

```suggestion
If you want to give it a spin, [you can download it now]({filename}/pages/download.md#stable) on Windows, macOS and Linux.
```
?

Reviewer 2: Yes, also nice.

This release brings a lot of new features and bugfixes:

Prepare your DJ set using [hotcue colors & labels]({filename}/news/2020-08-25-new-in-2-3-hotcue-colors.md), mark [intro/outro sections]({filename}/news/2020-07-09-intro-outro-sections.md) in your tracks and add [track colors]({filename}/news/2020-10-24-new-in-2-3-track-colors.md) for easier library navigation.
Our new multithreaded analysis and [more accurate key detection]({filename}/news/2021-04-08-new-in-2-3-keyfinder.md) will improve the preparation process further.

Mixxx 2.3.0 comes with a new default skin:
"LateNight" underwent a massive redesign and replaces "Deere" as default skin, so check out the [screenshots page]({filename}/pages/screenshots.md).

If you're a Rekordbox or Serato user, switching to Mixxx becomes a lot easier:
You can now play tracks directly from USB drives that contain [Rekordbox]({filename}/news/2020-07-20-new-in-2-3-rekordbox-support.md) and [Serato](2021-02-08-new-in-2-3-serato-support.md) libraries.

> We also added support for recording/streaming in the [Opus codec](https://opus-codec.org/), introduced [deck cloning]({filename}/news/2020-11-15-new-in-2-3-deck-clone.md), polished the library and preferences (including the controller workflow) and [switched to the CMake build system generator]({filename}/news/2020-12-14-scons-cmake-migration.md).

Reviewer 2: ```suggestion
We also added support for recording/streaming in the [Opus](https://opus-codec.org/) and [HE-AAC](https://www.iis.fraunhofer.de/en/ff/amm/broadcast-streaming/heaac.html) codecs, introduced [deck cloning]({filename}/news/2020-11-15-new-in-2-3-deck-clone.md), polished the library and preferences (including the controller workflow) and [switched to the CMake build system generator]({filename}/news/2020-12-14-scons-cmake-migration.md).
```

> We also added support for recording/streaming in the [Opus](https://opus-codec.org/) and [HE-AAC](https://www.iis.fraunhofer.de/en/ff/amm/broadcast-streaming/heaac.html) codecs, introduced [deck cloning]({filename}/news/2020-11-15-new-in-2-3-deck-clone.md), polished the library and preferences (including the controller workflow) and [switched to the CMake build system generator]({filename}/news/2020-12-14-scons-cmake-migration.md).

Reviewer 2: Can we move the CMake part to the infrastucture part below. Than we don't mix dev features with user features.

Reviewer 3: I agree. Apart from that this is a good read!

For the full list of changes, have a look at the [changelog](https://manual.mixxx.org/2.3/en/chapters/appendix/version_history.html) and the [2.3.0 milestone on Launchpad](https://launchpad.net/mixxx/+milestone/2.3.0).

This has been a rather big release, with lots of useful changes, bug fixes and improvements.

> All in all, a total of 7183 changes changed over 1 million lines of code were changed since the [2.2.4 release]({filename}/news/2020-06-11-Mixxx-2-2-4-released.md)!

Reviewer 3: ```suggestion
All in all, with a total of 7183 changes over 1 million lines of code were modified since the [2.2.4 release]({filename}/news/2020-06-11-Mixxx-2-2-4-released.md)!
```

**TODO:** Update the numbers above using `git shortlog -sn release-2.2.4..upstream/2.3 | wc -l`, `git log --oneline --no-merges release-2.2.4..upstream/2.3 -- . ':!lib' | wc -l` and `git diff --shortstat release-2.2.4..upstream/2.3 -- . ':!lib`.

Despite our plans to release Mixxx more often, it's been two years since the 2.2.x release, and 2.3 has been in the beta phase for almost a year now.
The reason for these delays is that we worked on some big refactorings and [infrastructure updates]({filename}/news/2021-02-23-build-infrastructure-updates.md) that were necessary and streamline our process, but also took a lot of time.
Our small development team can always need a helping hand, so if you want to help out and make Mixxx better, [get in touch with us]({filename}/pages/get-involved.md)!

> Many people spent their free time working on Mixxx and reported bugs, translated Mixxx into other languages, contributed controller mappings, improved skins and hacked on the core code. This release also features improvements to our manual made by our first intern from the Outreachy program. Thanks a lot to all our contributors, we really appreciate it!

Reviewer 3: ```suggestion
Many people spent their free time working on Mixxx and reported bugs, translated Mixxx into other languages, contributed controller mappings, improved skins and hacked on the core code. This release also features improvements to our manual made by our first intern from the [Outreachy program](https://www.outreachy.org/). Thanks a lot to all our contributors, we really appreciate it!
```

For the full list of changes, have a look at the [changelog](https://manual.mixxx.org/2.3/en/chapters/appendix/version_history.html) and the [2.3.0 milestone on Launchpad](https://launchpad.net/mixxx/+milestone/2.3.0).

> **Note:** KDE users may see broken icons due to [a bug in the kIconThemes 5.80 package](https://bugs.kde.org/show_bug.cgi?id=434451). If you're affected, please uninstall the package or update it to version 5.82. Unfortunately, not all Linux distributions have updated the package yet. For Ubuntu 21.04, a patched version of the package has been added to the [Mixxx PPA](https://launchpad.net/~mixxx/+archive/ubuntu/mixxx/+packages).

Reviewer 3: ```suggestion
**Note:** Linux users who see broken icons are affected by [a bug in the KDE kIconThemes 5.80 package](https://bugs.kde.org/show_bug.cgi?id=434451). If you don't have a newer version of that package in your repos, yet, see [this comment](https://bugs.launchpad.net/mixxx/+bug/1922966/comments/36) for how to fix the issue.
```
I'd just like it to be a bit shorter..

title: Mixxx 2.3.0 released
authors: Jan Holthuis
tags: 2.3, 2.3.0, release announcement

> status: draft

Reviewer: change draft status!

Author: This is taken care of by CI after merge.

comments: yes

We are proud to announce the [release of Mixxx 2.3.0](https://github.com/mixxxdj/mixxx/releases/tag/2.3.0)!
If you want to give it a spin, [you can download it now]({filename}/pages/download.md#stable) on Windows, macOS and Linux.

This release brings a lot of new features and bugfixes:

Prepare your DJ set using [hotcue colors & labels]({filename}/news/2020-08-25-new-in-2-3-hotcue-colors.md), mark [intro/outro sections]({filename}/news/2020-07-09-intro-outro-sections.md) in your tracks and add [track colors]({filename}/news/2020-10-24-new-in-2-3-track-colors.md) for easier library navigation.
Our new multithreaded analysis and [more accurate key detection]({filename}/news/2021-04-08-new-in-2-3-keyfinder.md) will improve the preparation process further.

Mixxx 2.3.0 comes with a new default skin:

> "LateNight" underwent a massive redesign and replaces "Deere" as default skin, so check out the [screenshots page]({filename}/pages/screenshots.md).

Reviewer: this image does not appear in the deploy preview

!["LateNight" is the new default Skin in Mixxx 2.3.0](https://mixxx.org/theme/images/2.3/screenshots/latenight-palemoon-3840x2160.png)

> If you're a Rekordbox or Serato user, switching to Mixxx becomes a lot easier:

Reviewer: "becomes" should be either "has become" or "is"

For the full list of changes, have a look at the [changelog](https://manual.mixxx.org/2.3/en/chapters/appendix/version_history.html) and the [2.3.0 milestone on Launchpad](https://launchpad.net/mixxx/+milestone/2.3.0).

> 

Reviewer: instead of "KDE users" just say "linux users" -- Even though I'm on GNOME I was bitten by this.

Author: I accidently reverted a change by @ronso0, fixed.

This has been a rather big release, with lots of useful changes, bug fixes and improvements.
All in all, with a total of 7477 changes over 1 million lines of code were modified since the [2.2.4 release]({filename}/news/2020-06-11-Mixxx-2-2-4-released.md)!

<!-- Numbers acquired using `git shortlog -sn release-2.2.4..upstream/2.3 | wc -l`, `git log --oneline --no-merges release-2.2.4..upstream/2.3 -- . ':!lib' | wc -l` and `git diff --shortstat release-2.2.4..upstream/2.3 -- . ':!lib`. -->

Despite our plans to release Mixxx more often, it's been two years since the 2.2.x release, and 2.3 has been in the beta phase for almost a year now.
The reason for these delays is that we [switched to the CMake build system generator]({filename}/news/2020-12-14-scons-cmake-migration.md), worked on some big refactorings and [infrastructure updates]({filename}/news/2021-02-23-build-infrastructure-updates.md) that were necessary to streamline our process, but also took a lot of time.

Our small development team can always need a helping hand, so if you want to help out and make Mixxx better, [get in touch with us]({filename}/pages/get-involved.md)!

> Unfortunately, we receive almost no C++ contributions from Windows or macOS developers despite our large user base.
> Our whole core development team uses Linux, which makes it extremely hard to maintain support for these systems.

Reviewer 3: ```suggestion
Unfortunately, we receive almost no C++ contributions from Windows or macOS developers despite our large user base on those systems.
Our whole core development team uses Linux, which makes it extremely hard to maintain support for other systems.
```

Without significantly more contributions, the future of Mixxx on Windows and macOS is at stake.

> If you know a bit of C++ and use Windows or macOS, please consider helping out.

Author: @mixxxdj/developers Please check if this is okay or if something should be changed.

Many people spent their free time working on Mixxx and reported bugs, translated Mixxx into other languages, contributed controller mappings, improved skins and hacked on the core code.
This release also features improvements to our manual made by our first intern from the [Outreachy program](https://www.outreachy.org/).
Thanks a lot to all our contributors, we really appreciate it!

The Mixxx code (including skins and controller mappings) has received contributions by almost 100 people:

>     $ git log --pretty="format:%an" release-2.2.4..upstream/2.3 | sort -u

Reviewer 2: This list looks IMHO too hacky for a user targeted post. It contains duplicates and and is not cleaned up in terms of formatting.  Can't we just copy over the contributor list from the about box? 
Or just delete it?

Author: I thought showing the git command gives us developer street credibility, but ok :wink:

Author: Hmm, the dlgabout list lists developers from the last 2 years, not this release. and I think there are some minor contributors missing. Should I replace it anyway? I'd prefer to keep the current list, but I don't really care.

Reviewer 2: OK
